plugins/lsp/language-servers/nixd.nix

```diff
@@ -10,52 +10,52 @@ let
-  options.plugins.lsp.servers.nixd.settings = {
-    # The evaluation section, provide auto completion for dynamic bindings.
-    eval = {
-      target = {
-        args = helpers.defaultNullOpts.mkNullable (with types; listOf str) "[]" ''
-          Accept args as "nix eval".
-        '';
-
-        installable = helpers.defaultNullOpts.mkStr "" ''
-          "nix eval"
-        '';
-      };
-
-      depth = helpers.defaultNullOpts.mkInt 0 "Extra depth for evaluation";
-
-      workers = helpers.defaultNullOpts.mkInt 3 "The number of workers for evaluation task.";
-    };
-
+  # Options:
+  #  - https://github.com/nix-community/nixd/blob/main/nixd/docs/configuration.md
+  #  - https://github.com/nix-community/nixd/blob/main/nixd/include/nixd/Controller/Configuration.h
+  options = {
     formatting = {
-      command = helpers.defaultNullOpts.mkStr "nixpkgs-fmt" ''
+      command = helpers.defaultNullOpts.mkListOf types.str ''[ "nixpkgs-fmt" ]'' ''
         Which command you would like to do formatting.
-        Explicitly set to `"nixpkgs-fmt"` will automatically add `pkgs.nixpkgs-fmt` to the nixvim
+        Explicitly set to `["nixpkgs-fmt"]` will automatically add `pkgs.nixpkgs-fmt` to the nixvim
         environment.
       '';
     };
 
-    options = {
-      enable = helpers.defaultNullOpts.mkBool true ''
-        Enable option completion task.
-        If you are writing a package, disable this
+    options =
+      let
+        provider = types.submodule {
+          options = {
+            expr = mkOption {
+              type = types.str;
+              description = "Expression to eval. Select this attrset as eval .options";
+            };
+          };
+        };
+      in
+      helpers.mkNullOrOption (with helpers.nixvimTypes; attrsOf (maybeRaw provider)) ''
+        Tell the language server your desired option set, for completion.
+        This is lazily evaluated.
       '';
 
-      target = {
-        args = helpers.defaultNullOpts.mkNullable (with types; listOf str) "[]" ''
-          Accept args as "nix eval".
-        '';
-
-        installable = helpers.defaultNullOpts.mkStr "" ''
-          "nix eval"
-        '';
-      };
-    };
+    nixpkgs =
+      let
+        provider = types.submodule {
+          options = {
+            expr = mkOption {
+              type = types.str;
+              description = "Expression to eval. Treat it as `import <nixpkgs> { }`";
+            };
+          };
+        };
+      in
+      helpers.mkNullOrOption (helpers.nixvimTypes.maybeRaw provider) ''
+        This expression will be interpreted as "nixpkgs" toplevel
+        Nixd provides package, lib completion/information from it.
+      '';
   };
 
   config = mkIf cfg.enable {
-    extraPackages = optional (cfg.settings.formatting.command == "nixpkgs-fmt") pkgs.nixpkgs-fmt;
+    extraPackages = optional (cfg.settings.formatting.command == [ "nixpkgs-fmt" ]) pkgs.nixpkgs-fmt;
```

Comment:
Maybe not on scope for this PR, but it'd be nice to do this for other nix formatters too. Maybe reference an attr set mapping name to package, then do `extraPackages = set.${cfg.settings.formatting.command} or null;`, although that doesn't work for lists...

Reply:
Yeah definitively, I didn't want to clutter this PR with that